### PR TITLE
various python package version and dependency updates

### PR DESCRIPTION
**Pull request by lee218llnl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-logilab-common/package.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ class PyLogilabCommon(Package):
 
     version('1.2.0', 'f7b51351b7bfe052746fa04c03253c0b')
 
-    extends("python")
+    extends('python', ignore=r'bin/pytest')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-logilab-common/package.py]: What does this ignore notation mean?  I've never seen it before...

File: var/spack/repos/builtin/packages/py-pylint/package.py [hunk 2]
```diff
@@ -33,7 +33,10 @@ class PyPylint(Package):
     version('1.4.1', 'df7c679bdcce5019389038847e4de622')
     version('1.4.3', '5924c1c7ca5ca23647812f5971d0ea44')
 
-    extends('python')
+    extends('python', ignore=r'bin/pytest')
+    depends_on('py-six', type=nolink)
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-pylint/package.py]: Can you use `type=('build','run')`?  There is a desire to deprecate `nolink`.

File: var/spack/repos/builtin/packages/py-pyside/package.py [hunk 3]
```diff
@@ -38,6 +38,7 @@ class PyPyside(Package):
 
     extends('python')
     depends_on('py-setuptools', type='build')
+    depends_on('py-sphinx')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-pyside/package.py]: Why is Sphinx need to build Python bindings for Qt4?  I assume this is for PySide documentation?  Can you make it optional, under a variant called `+doc`?

File: var/spack/repos/builtin/packages/py-pytables/package.py [hunk 4]
```diff
@@ -32,6 +32,7 @@ class PyPytables(Package):
     url      = "https://github.com/PyTables/PyTables/archive/v.3.2.2.tar.gz"
 
     version('3.2.2', '7cbb0972e4d6580f629996a5bed92441')
+    version('3.3.0', '056c161ae0fd2d6e585b766adacf3b0b', url='https://github.com/PyTables/PyTables/archive/v3.3.0.tar.gz')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-pytables/package.py]: Can the URL go on the next line?
Also... 3.3.0 need to come before 3.2.2

File: var/spack/repos/builtin/packages/py-pyside/package.py (hunk 3 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyside/package.py]: `type='nolink'`?

File: var/spack/repos/builtin/packages/py-pytables/package.py (hunk 4 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytables/package.py]: Is the URL necessary? If so we may want to add this to the url-parsing. @tgamblin 

Also, newer versions should go first.

File: var/spack/repos/builtin/packages/texlive/package.py [hunk 5]
```diff
@@ -41,7 +41,7 @@ class Texlive(Package):
     # digest values, but don't be surprised if this package is
     # briefly unbuildable.
     #
-    version('live', '01461ec2cc49fe0b14812eb67abbea46',
+    version('live', 'ad230fa814d122084c13d75c0b135fda',
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/texlive/package.py]: Can you remove this TeXLive update from this PR?  It is already in #2746

File: var/spack/repos/builtin/packages/zlib/package.py [hunk 6]
```diff
@@ -33,6 +33,7 @@ class Zlib(AutotoolsPackage):
     homepage = "http://zlib.net"
     url = "http://zlib.net/zlib-1.2.8.tar.gz"
 
+    version('1.2.10', 'd9794246f853d15ce0fcbf79b9a3cf13')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/zlib/package.py]: Please remove `zlib` from this PR, it was just merged in #2735

File: var/spack/repos/builtin/packages/python/package.py [hunk 7]
```diff
@@ -44,6 +44,7 @@ class Python(Package):
     list_url = "https://www.python.org/downloads/"
     list_depth = 2
 
+    version('3.6.0', '3f7062ccf8be76491884d0e47ac8b251')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/python/package.py]: Yay, Python 3.6 is out!

File: var/spack/repos/builtin/packages/py-logilab-common/package.py (hunk 1 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-logilab-common/package.py]: It means "don't symlink that particular file when you run `spack activate py-logilab-common`". It is documented [here](http://spack.readthedocs.io/en/latest/packaging_guide.html#extensions)

File: var/spack/repos/builtin/packages/py-pyside/package.py (hunk 3 above)
Comment (lee218llnl) [inline var/spack/repos/builtin/packages/py-pyside/package.py]: yup, will fix

File: var/spack/repos/builtin/packages/py-pytables/package.py (hunk 4 above)
Comment (lee218llnl) [inline var/spack/repos/builtin/packages/py-pytables/package.py]: yes, this was necessary to get it to build.

File: var/spack/repos/builtin/packages/py-pylint/package.py (hunk 2 above)
Comment (lee218llnl) [inline var/spack/repos/builtin/packages/py-pylint/package.py]: If we want to take this route, I think we will need to do a sweep of all the packages to do this, so I probably won't do this in this PR.

File: var/spack/repos/builtin/packages/py-pyside/package.py (hunk 3 above)
Comment (lee218llnl) [inline var/spack/repos/builtin/packages/py-pyside/package.py]: The build will actually fail if sphinx-build is not found.

File: var/spack/repos/builtin/packages/py-pytables/package.py (hunk 4 above)
Comment (lee218llnl) [inline var/spack/repos/builtin/packages/py-pytables/package.py]: will do

File: var/spack/repos/builtin/packages/texlive/package.py (hunk 5 above)
Comment (lee218llnl) [inline var/spack/repos/builtin/packages/texlive/package.py]: will do

File: var/spack/repos/builtin/packages/zlib/package.py (hunk 6 above)
Comment (lee218llnl) [inline var/spack/repos/builtin/packages/zlib/package.py]: will do

**Comment by tgamblin** [comment]
@lee218llnl: #2320 just went in, FYI.  It may contain other/conflicting packages

**Comment by lee218llnl** [comment]
@tgamblin with a cursory glance, I think we're OK. We'll find out soon though.

File: var/spack/repos/builtin/packages/py-pytables/package.py (hunk 4 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytables/package.py]: Oh, just noticed that one is `v3.3.0` and the other is `v.3.2.2` with an extra dot at the beginning. Yep, no other way around that.

**Comment by tgamblin** [comment]
Thanks!

**Pull request merged**